### PR TITLE
[jiterator] sqrt-rsqrt : complex

### PR DESCRIPTION
**Pull request by kshitij12345**
As per title

Relies on existing OpInfo tests from `test_ops.py` and `test_unary_ufuncs.py`

### CODE REVIEW

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu [hunk 1]
```diff
@@ -53,19 +55,46 @@ void expm1_kernel_cuda(TensorIteratorBase& iter) {
...
 void rsqrt_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+  auto common_dtype = iter.common_dtype();
+  if (at::isComplexType(common_dtype)) {
+    #if AT_USE_JITERATOR()
+      static const auto rsqrt_string = jiterator_stringify(
+          template <typename T>
+          T rsqrt_kernel(T x) {
+            const T one = T{1};
+            // std::sqrt for c10::complex is overloaded in c10/util/complex_math.h
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/UnaryOpsKernel.cu:78]: Shouldn't the complex headers the jiterator includes define `std::sqrt`? I thought that file worked by extending the standard namespace with the complex math operations

cc @zasdfgbnm

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu [hunk 2]
```diff
@@ -74,14 +103,40 @@ void rsqrt_kernel_cuda(TensorIteratorBase& iter) {
...
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.common_dtype(), "sqrt_cuda", [&]() {
-    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
-      return ::sqrt(a);
+  auto common_dtype = iter.common_dtype();
+  if (at::isComplexType(common_dtype)) {
+    #if AT_USE_JITERATOR()
+      static const auto sqrt_string = jiterator_stringify(
+          template <typename T>
+          T sqrt_kernel(T x) {
+            return sqrt(x);
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/UnaryOpsKernel.cu:117]: Same question about `std::sqrt` here

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu (hunk 1 above)
Comment (zasdfgbnm) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu:78]: It should: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/llvm_complex.cpp#L727

Comment (kshitij12345) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: I think it worked out due to argument dependent lookup.

Have qualified the call with `std` namespace.

**Comment by mruberry** [comment]
Yikes -- 30 job failures is a lot! This might require a rebase -- at least some of them appear to be due to unrelated build issues

**Comment by kshitij12345** [comment]
@mruberry build failures are fixed with latest master. PTAL :)

**Review by anjali411 — APPROVED** [review]
LGTM!

**Pull request closed without merging**